### ISSUE TITLE
Feature Engineering

### ISSUE THREAD

**Issue by meera998** [meera998/Auto_trader#4]
Feature engineering is about creating new feature or transforming existing features from one representation to another which enhance our machine learning model.
Since our target is numeric and we are using regression model for prediction, we have to transform categorical features in our dataset.
Because we have high cardinality in our categorical features, we will use Target Encoder to avoid high dimensionality in our dataset which may lead to overfitting.
A major issue emerges when applying Target Encoder which is data leakage. Data leakage occurs when information outside of the training set is used to build the model, So the model can not generalize properly. To avoid this, we split our data into training and test set before applying Target Encoder.
